AgnosticUI/agnosticui
# Add storybook to Svelte TS project

# Pull Request Template

## Description

This PR adds Storybook with Typescript support to the `agnostic-svelte-ts` project:
- Stories can now use the [Svelte native format](https://storybook.js.org/blog/storybook-for-svelte/#svelte-native-format
)
- The `copystyles.cjs` file and the npm script invoking it is copied over to the project, but currently throws an error because of missing components
- Prettier config is updated to use the SvelteKit defaults as well as settings from `agnostc-svelte`
- The Alert component is copied with some type errors fixed and the new story format

Related to #127

## Testing steps

```
cd agnostic-svelte-ts
npm i && npm run storybook
```

Storybook should build all stories with TS sources and common CSS successfully:
<img width="1509" alt="image" src="https://user-images.githubusercontent.com/11555574/164965930-c9f58e52-d124-465f-900e-8efff3870b79.png">

## Checklist:

Please delete options that are not relevant.

- [X] Have you submitted new feature or bugfix issue? Unless it's a small bugix, we generally prefer to track PRs against an issue which starts a dialogue to give us context.

_We realize above checklist is pretty intimidating, reach out with an at mention on your issue if you're interested in contributing but need some clarifications!_
